src/services/widgets/widgets.svc.coffee

```diff
@@ -74,22 +74,39 @@ angular
         setting.initialize()
       return true
 
-    @updateWidgetSettings = (widget, needContentReload=true) ->
+    @updateWidgetSettings = (widget, needContentReload=true, ignoreReach=false) ->
       widget.isEditMode = false
-
       if _.isEmpty(widget.settings)
         $log.warn("ImpacWidgetsSvc: Tried to update widget: #{widget.id} with no settings", widget)
         return false
 
+      changedGlobalSetting = _.find widget.settings, (setting)-> setting.reach == 'dashboard'
+
+      if (changedGlobalSetting && !ignoreReach)
+        return _self.updateAllSameWidgets(widget, ImpacDashboardsSvc.getCurrentDashboard(), changedGlobalSetting)
+
       widget.isLoading = true if needContentReload
       meta = _.reduce(_.map(widget.settings, (set) -> set.toMetadata() ), (result, setMeta) -> angular.merge(result, setMeta))
-      
-      return _self.update(widget, { metadata: meta }).then(
+
+      _self.update(widget, { metadata: meta }).then(
         (updatedWidget) ->
           if needContentReload
             _self.show(updatedWidget).finally( -> updatedWidget.isLoading = false )
       )
 
+    @updateAllSameWidgets = (widget, dashboard, settings) ->
```

Comment:
My bad for 1.: I didn't see the recursive inclusion issue. Let's just add a comment `TODO: move logic in ImpacDashboardsSvc`, I'll try to figure out a way to move it later.

For 2.: I get your point, but actually, "rewrite it in all widgets were the setting key is defined" is basically what we are trying to do here: rewrite all the widgets were the user defined a list of lead statuses (for leads), or of sales stage statuses (for opportunities). So, that's why the setting key has to be specific to the information held (regarding the other comment).